src/qgis_stac/gui/assets_dialog.py

```diff
@@ -412,6 +412,18 @@ def sign_asset_href(self, asset_href):
             signed_href = pc.sign(asset_href)
             return signed_href
 
+        if asset_href.startswith("s3://"):
```

Comment:
@ahuarte47 can you put the code for with the new changes into its own function the `sign_asset_href` method is for signing planetary computer SAS based items hrefs with a SAS token.

Reply:
Lets make another function for the new changes.

Reply:
@Samweli can't really understand. The plugin and the sign method are not named as microsoft or planetarycomputer dependent => because the plugin is a Common, have to be and remain general.

`sign_asset_href` is a enough general name that can refer to the signing action that is common to any cloud resource not only that from microsoft services.

In resume, what is the rationale to move a sign feature for only GC or AWS to a different method respect signing of microsoft cloud?